### PR TITLE
[JBPM-9277]  JMS Audit Data Replication in spring boot

### PR DESCRIPTION
**Pull request by elguardian**
refactor task listeners and process listener to adapters
for data replication

### CODE REVIEW

**Comment by elguardian** [comment]
depends on https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1431

**Comment by elguardian** [comment]
Jenkins retest this please.

File: jbpm-audit/src/main/java/org/jbpm/process/audit/NodeInstanceLog.java [hunk 1]
```diff
@@ -99,7 +99,7 @@ public long getId() {
     	return id;
     }
     
-    void setId(long id) {
+    public void setId(long id) {
```
[the remark this answers is not in the document]
Comment (elguardian) [reply jbpm-audit/src/main/java/org/jbpm/process/audit/NodeInstanceLog.java:102]: validation fails in jms receiver if those aren't public. Anyways just for coherence. (I override the seq generator)

File: jbpm-audit/src/main/java/org/jbpm/process/audit/ProcessInstanceLog.java [hunk 2]
```diff
@@ -132,7 +132,7 @@ public long getId() {
     	return id;
     }
     
-    void setId(long id) {
+    public void setId(long id) {
```
[the remark this answers is not in the document]
Comment (elguardian) [reply jbpm-audit/src/main/java/org/jbpm/process/audit/ProcessInstanceLog.java:135]: validation fails in jms receiver if those aren't public. Anyways just for coherence. (I override the seq generator)

File: jbpm-audit/src/main/java/org/jbpm/process/audit/AbstractAuditLoggerAdapter.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    public static final String METADATA_VARIABLEINSTANCE_LOG = "VariableInstanceLog";
+
+    public AbstractAuditLoggerAdapter() {
+
+    }
+    public AbstractAuditLoggerAdapter(WorkingMemory workingMemory) {
+        super(workingMemory);
+    }
+
+    public void setProcessInstanceMetadata(org.kie.api.runtime.process.ProcessInstance pi, String key, Object value) {
```
[the remark this answers is not in the document]
Comment (elguardian) [reply jbpm-audit/src/main/java/org/jbpm/process/audit/AbstractAuditLoggerAdapter.java:42]: done

File: jbpm-audit/src/main/java/org/jbpm/process/audit/AbstractAuditLoggerAdapter.java [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+        ((ProcessInstanceImpl) pi).getMetaData().put(key, value);
+    }
+
+    public Object getProcessInstanceMetadata(org.kie.api.runtime.process.ProcessInstance pi, String key) {
+        return ((ProcessInstanceImpl) pi).getMetaData().get(key);
+    }
+
+    public void setNodeInstanceMetadata(org.kie.api.runtime.process.NodeInstance pi, String key, Object value) {
```
[the remark this answers is not in the document]
Comment (elguardian) [reply jbpm-audit/src/main/java/org/jbpm/process/audit/AbstractAuditLoggerAdapter.java:50]: done

File: jbpm-audit/src/main/java/org/jbpm/process/audit/AbstractAuditLoggerAdapter.java [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+        ((NodeInstanceImpl) pi).getMetaData().put(key, value);
+    }
+
+    public Object getNodeInstanceMetadata(org.kie.api.runtime.process.NodeInstance pi, String key) {
```
[the remark this answers is not in the document]
Comment (elguardian) [reply jbpm-audit/src/main/java/org/jbpm/process/audit/AbstractAuditLoggerAdapter.java:54]: done

File: jbpm-audit/src/main/java/org/jbpm/process/audit/JPAWorkingMemoryDbLogger.java [hunk 6]
```diff
@@ -57,18 +56,19 @@
...
+
     private boolean isJTA = true;
     private boolean sharedEM = false;
-    
+
     private EntityManagerFactory emf;
-    
-    private ProcessIndexerManager indexManager = ProcessIndexerManager.get();
+
+    protected ProcessIndexerManager indexManager = ProcessIndexerManager.get();
```
[the remark this answers is not in the document]
Comment (elguardian) [reply jbpm-audit/src/main/java/org/jbpm/process/audit/JPAWorkingMemoryDbLogger.java:70]: removed

File: jbpm-human-task/jbpm-human-task-audit/src/main/java/org/jbpm/services/task/audit/JPATaskLifeCycleEventListener.java [hunk 7]
```diff
@@ -48,7 +48,11 @@
  *
  */
 public class JPATaskLifeCycleEventListener extends PersistableEventListener implements TaskLifeCycleEventListener {
-    
+
+    public static final String METADATA_TASK_EVENT = "TASK_EVENT";
+    public static final String METADATA_AUDIT_TASK = "TASK_AUDIT_EVENT";
+    public static final String METADATA_INPUT_VAR_EVENT = "TASK_INPUT_VAR_EVENT";
```
[the remark this answers is not in the document]
Comment (elguardian) [reply jbpm-human-task/jbpm-human-task-audit/src/main/java/org/jbpm/services/task/audit/JPATaskLifeCycleEventListener.java:54]: it should be reference by indexAndPersistVariables method

File: jbpm-audit/src/main/java/org/jbpm/process/audit/JPAWorkingMemoryDbLogger.java [hunk 8]
```diff
@@ -107,154 +107,124 @@ public JPAWorkingMemoryDbLogger(Environment env) {
         internalSetIsJTA(env);
     }
 
-    private void internalSetIsJTA(Environment env) { 
+    protected void internalSetIsJTA(Environment env) {
```
[the remark this answers is not in the document]
Comment (elguardian) [reply jbpm-audit/src/main/java/org/jbpm/process/audit/JPAWorkingMemoryDbLogger.java:110]: done

File: jbpm-audit/src/main/java/org/jbpm/process/audit/JPAWorkingMemoryDbLogger.java [hunk 9]
```diff
@@ -107,154 +107,124 @@ public JPAWorkingMemoryDbLogger(Environment env) {
...
         Boolean bool = (Boolean) env.get("IS_JTA_TRANSACTION");
         if (bool != null) {
-        	isJTA = bool.booleanValue();
+            isJTA = bool.booleanValue();
```
[the remark this answers is not in the document]
Comment (elguardian) [reply jbpm-audit/src/main/java/org/jbpm/process/audit/JPAWorkingMemoryDbLogger.java:113]: removed the change.

File: jbpm-audit/src/main/java/org/jbpm/process/audit/AbstractAuditLoggerAdapter.java [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+    }
+    public AbstractAuditLoggerAdapter(WorkingMemory workingMemory) {
+        super(workingMemory);
+    }
+
+    public void setProcessInstanceMetadata(org.kie.api.runtime.process.ProcessInstance pi, String key, Object value) {
+        ((ProcessInstanceImpl) pi).getMetaData().put(key, value);
+    }
+
+    public Object getProcessInstanceMetadata(org.kie.api.runtime.process.ProcessInstance pi, String key) {
```
[the remark this answers is not in the document]
Comment (elguardian) [reply jbpm-audit/src/main/java/org/jbpm/process/audit/AbstractAuditLoggerAdapter.java]: done

**Review by afalhambra — APPROVED** [review]
Nice work, looks good to me.

**Pull request merged**